### PR TITLE
adding error handling to main enpoints

### PR DESCRIPTION
**Pull request by pariosur**
- Handling errors with try / except 
- Using logging to 'print' the messages

### CODE REVIEW

File: src/metal_sdk/metal.py [hunk 1]
```diff
@@ -145,10 +187,17 @@ def delete_one(self, id: str, index_id=None):
...
+                res = self.request("delete", url)
+                res.raise_for_status()
+                return None
+        except httpx.HTTPStatusError as e:
+            response_data = e.response.json()
+            status_code = e.response.status_code
+            error_message = response_data.get('error', {}).get('message') or response_data.get('message', f"HTTP {status_code} error")
+            formatted_error = f"\n{'-'*50}\nError occurred while accessing {url}: {error_message}\n{'-'*50}\n"
+            logger.exception(formatted_error)
+            return response_data
```
Comment (softboyjimbo) [inline src/metal_sdk/metal.py:200]: nice - i wonder if we can consolidate this logic into a single method that can be used by all of these separate methods? perhaps something like:

```python
def fetch(self, method, url, payload):
    try:
        res = self.request("delete", url)
        res.raise_for_status()
        return res
    except httpx.HTTPStatusError as e:
            response_data = e.response.json()
            status_code = e.response.status_code
            error_message = response_data.get('error', {}).get('message') or response_data.get('message', f"HTTP {status_code} error")
            formatted_error = f"\n{'-'*50}\nError occurred while accessing {url}: {error_message}\n{'-'*50}\n"
            logger.exception(formatted_error)
            return response_data
```

Then you can make requests from this method like:

```python
res = self.fetch("delete", url)
return res
```

Comment (pariosur) [reply src/metal_sdk/metal.py]: awesome thanks for this!

**Review by softboyjimbo — APPROVED** [review]
@pariosur this looks great! thanks for the updates

**Pull request merged**